naga/src/front/wgsl/parse/mod.rs

```diff
@@ -261,6 +261,21 @@ impl<'a> BindingParser<'a> {
     }
 }
 
+/// Configuration for the whole parser run.
+pub struct ParserOptions {
+    /// Controls whether the parser should parse comments.
+    pub parse_comments: bool,
+}
+
+impl ParserOptions {
```

Comment:
This should just be named `Options`; the fact that it's from `naga::front::wgsl` makes it clear that it contains parsing options.

Rather than having a `new` method, `Options` structs should implement `Default`.

Reply:
Good point about prefix being redundant 👍

I can implement default, but it's not const so this was complaining from an existing caller.

Reply:
Thanks for that information.

I don't think `const` was added without reason here, it goes back to https://github.com/gfx-rs/wgpu/commit/6035b07b786135f65cb4140f99cf67cc0381b4c3#diff-46894d640690ef35dff3deaf0bb5416be6e4d00eaba890e3181628d769f6f413R1735 ; and the commit message has a `Fixes https://github.com/gfx-rs/wgpu/pull/2071: Global item does not support 'const'`.

I'd prefer to keep this PR as close to existing code as possible.